File: backend_Myecoprint/backend_myecoprint/models/company.js

const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Company = sequelize.define('Company', {
  company_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true,
  },
  company_name: {
    type: DataTypes.STRING,
  },
  company_type: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  password: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  image: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  contact: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  createdAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  updatedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  deletedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
}, {
  timestamps: false, 
  paranoid: true, 
  tableName: 'companies', 
});

module.exports = Company;

File: Myecoprint_admin/ncej_admin/lib/data/module/company.dart

class Company {
  final int companyId;
  final String companyName;
  final String companyType;
  final String password;
  final String image;
  final String contact;

  Company({
    required this.companyId,
    required this.companyName,
    required this.companyType,
    required this.password,
    required this.image,
    required this.contact,
  });

  factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
      companyId: json['company_id'],
      companyName: json['company_name'],
      companyType: json['company_type'],
      password: json['password'],
      image: json['image'],
      contact: json['contact'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'company_id': companyId,
      'company_name': companyName,
      'company_type': companyType,
      'password': password,
      'image': image,
      'contact': contact,
    };
  }
}
